src/Cli/dotnet/commands/dotnet-workload/install/WorkloadInstallCommand.cs

```diff
@@ -68,67 +70,100 @@ private void ValidateWorkloadIdsInput()
         public override int Execute()
         {
             bool usedRollback = !string.IsNullOrWhiteSpace(_fromRollbackDefinition);
-            if (_printDownloadLinkOnly)
+
+            WorkloadHistoryRecorder recorder = new WorkloadHistoryRecorder(_workloadResolver, _workloadInstaller);
+            recorder.HistoryRecord.CommandName = IsRunningRestore ? "restore" : "install";
```

Comment:
I see these manual strings being inserted into this CommandName property. Would it make sense to use an enum instead with the different types of commands?

Reply:
Maybe...personally, I think I'd vote against this because although there aren't that many possibilities, we don't touch it much (it's basically set once, then used once), and when we ultimately use it, it needs to be a string for printing, so that would add a little more indirection for a little more tidiness...

Certainly a reasonable suggestion; I just prefer a string here, personally.

Reply:
The purpose of using an enum is it consolidates all the possible values to one location. So, for me to find the values possible, I had to jump around to all uses of `CommandName` in your PR and keep track of the values. I saw:
- `install`
- `uninstall`
- `restore`
- `update`

If the only values applicable for `CommandName` is a closed set (meaning, not user-inputable or other external names), then using an enum is preferred. It lets every dev know that it is intended to be a finite set of values.

Here's a quick mockup showing how easy it is. `ToString` on an enum value in C# outputs the name of the value, not the backing integer. https://dotnetfiddle.net/AuISsD

![image](https://github.com/dotnet/sdk/assets/17788297/c1b215f0-3a3f-4eec-b64a-59a92cfaf8dc)

Shows this output:
![image](https://github.com/dotnet/sdk/assets/17788297/6817498c-ac96-4b68-8f1b-7a6985a90d6c)